jgrapht-core/src/main/java/org/jgrapht/alg/matching/MaximumWeightBipartiteMatching.java

```diff
@@ -17,39 +17,71 @@
  */
 package org.jgrapht.alg.matching;
 
-import java.util.*;
+import java.math.BigDecimal;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
 
-import org.jgrapht.*;
-import org.jgrapht.alg.interfaces.*;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphTests;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.MatchingAlgorithm;
+import org.jgrapht.util.GenericFibonacciHeap;
 
 /**
- * This class finds a maximum weight matching of a simple undirected weighted bipartite graph. The
- * algorithm runs in O(V|E|^2). The algorithm is described in The LEDA Platform of Combinatorial and
- * Geometric Computing, Cambridge University Press, 1999.
- * https://people.mpi-inf.mpg.de/~mehlhorn/LEDAbook.html Note: the input graph must be bipartite
- * with positive integer edge weights
- *
+ * Maximum weight matching in bipartite graphs.
+ * 
+ * <p>
+ * Running time is $O(n(m+n \log n))$ where n is the number of vertices and m the number of edges of the
+ * input graph. Uses exact arithmetic and produces a certificate of optimality in the form of a
+ * tight vertex potential function.
+ * 
+ * <p>
+ * This is the algorithm described in the
+ * <a href="https://people.mpi-inf.mpg.de/~mehlhorn/LEDAbook.html">LEDA book</a>. See the LEDA
+ * Platform of Combinatorial and Geometric Computing, Cambridge University Press, 1999.
+ * 
  * @param <V> the graph vertex type
  * @param <E> the graph edge type
  *
- * @author Graeme Ahokas
+ * @author Dimitrios Michail
  */
 public class MaximumWeightBipartiteMatching<V, E>
     implements MatchingAlgorithm<V, E>
 {
     private final Graph<V, E> graph;
-    private Set<V> partition1;
-    private Set<V> partition2;
+    private final Set<V> partition1;
+    private final Set<V> partition2;
+
+    private final Comparator<BigDecimal> comparator;
+
+    // vertex potentials
+    private Map<V, BigDecimal> pot;
+    // the matched edge of a vertex, also used to check if a vertex is free
+    private Map<V, E> matchedEdge;
 
-    private Map<V, Long> vertexWeights;
-    private Map<V, Boolean> hasVertexBeenProcessed;
-    private Map<E, Boolean> isEdgeMatched;
+    // shortest path related data structures
+    private GenericFibonacciHeap<BigDecimal, V> heap;
+    private Set<V> isInHeap;
```

Comment:
Is it possible to use nodeInHeap to compute isInHeap as well (to avoid redundancy)?

Reply:
Indeed, nice suggestion.